src/imports.rs

```diff
@@ -244,9 +244,65 @@ pub(crate) fn normalize_use_trees_with_granularity(
             }
         }
     }
+
+    if import_granularity == ImportGranularity::ModuleCondensed {
+        condense_use_trees(&mut result);
+    }
+
     result
 }
 
+/// Merge singleton imports into parent modules' use-trees where it does't
+/// introduce nested braces.
+fn condense_use_trees(use_trees: &mut Vec<UseTree>) {
+    // Sort by path length to put singletons after the trees into which they
+    // may be absorbed.
+    use_trees.sort_unstable_by_key(|tree| tree.path.len());
+
+    // Search for singletons back-to-front to preserve the just mentioned
+    // relative order between absorbing trees and singletons. Note that a
+    // singleton that is found and merged into _some_ tree is `swap_remove`d
+    // from the `use_trees` vector.
+    for n in (0..use_trees.len()).rev() {
+        let singleton = &use_trees[n];
+
+        if !singleton.is_singleton() || singleton.attrs.is_some() || singleton.contains_comment() {
+            continue;
+        }
+
+        let mut best_index_and_len = None;
+
+        // Search front-to-back for a tree to merge the singleton into. If
+        // multiple trees are equally good candidates, prefer the earliest one.
```

Comment:
Could you add this as a unit test/comment the part in the ui test that is exercising the "prefer the earliest one" logic?